### PR TITLE
Ref #9055 - make packages depended on, a configurable parameter

### PR DESCRIPTION
**Pull request by dustints**

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # Katello Install
 class katello::install {
-  package { $katello::package_names:
-    ensure => installed,
+
+  if $katello::packages == undef {
```
Comment (ekohl) [inline manifests/install.pp]: You can also solve this using [pick](https://github.com/puppetlabs/puppetlabs-stdlib#functions):
```puppet
$packages = pick($katello::packages, $katello::package_names)
package { $packages:
  ensure => installed,
}
```

File: manifests/init.pp [hunk 2]
```diff
@@ -54,6 +56,8 @@
   $proxy_username = $katello::params::proxy_username,
   $proxy_password = $katello::params::proxy_password,
   $cdn_ssl_version = $katello::params::cdn_ssl_version,
+
+  $packages = $katello::params::packages
```
Comment (ekohl) [inline manifests/init.pp]: Why introduce a separate parameter when there's already one in params that you could expose here?

**Comment by dustints** [comment]
@ekohl addressed...just expose package_names

**Comment by ekohl** [comment]
:+1: pending tests.

**Comment by dustints** [comment]
@ehelms saw your comment in https://github.com/Katello/katello/pull/4923.
is this ok to merge?

**Comment by ehelms** [comment]
I want to say yes, but I have been intentionally testing the entire stack and process to ensure that all proposed changes solve the problem.

**Comment by ehelms** [comment]
ACK

File: manifests/init.pp [hunk 3]
```diff
@@ -54,6 +56,8 @@
   $proxy_username = $katello::params::proxy_username,
   $proxy_password = $katello::params::proxy_password,
   $cdn_ssl_version = $katello::params::cdn_ssl_version,
+
+  $package_names = $katello::params::package_names
```
Comment (ekohl) [inline manifests/init.pp]: Nitpick: I tend to prefer trailing commas because they lead to smaller differs.

Comment (dustints) [inline manifests/init.pp]: @ekohl addressed

Comment (ehelms) [inline manifests/init.pp]: @dustints you might have forgot to push your changes

Comment (dustints) [inline manifests/init.pp]: @ehelms darn! yah that was the case...ok pushed this time.

**Comment by ehelms** [comment]
APT

**Comment by ekohl** [comment]
:+1:

**Pull request merged**